src/main/kotlin/com/exactpro/th2/lwdataprovider/db/CradleMessageExtractor.kt

```diff
@@ -157,36 +158,37 @@ class CradleMessageExtractor(
         val remaining: LinkedList<StoredMessage> = LinkedList()
         while (iterator.hasNext()) {
             val measurement = dataMeasurement.start("process_cradle_group_batch")
+            @Suppress("ConvertTryFinallyToUseCall")
             try {
                 sink.canceled?.apply {
                     logger.info { "canceled because: $message" }
                     return
                 }
                 prev = currentBatch
                 currentBatch = iterator.next()
-                check(prev.firstTimestamp <= currentBatch.firstTimestamp) {
-                    "Unordered batches received: ${prev.toShortInfo()} and ${currentBatch.toShortInfo()}"
+                check(orderStrategy.checkBatchSequence(prev, currentBatch)) {
+                    "Unordered batches received for $orderStrategy: ${prev.toShortInfo()} and ${currentBatch.toShortInfo()}"
                 }
                 val needFiltration = prev.isNeedFiltration()
 
-                if (prev.lastTimestamp < currentBatch.firstTimestamp) {
+                if (orderStrategy.checkBatchContact(prev, currentBatch)) {
```

Comment:
If the batches cannot overlap this should never happen, right? Probably, we should report an error here...